ocaml/lambda/lambda.ml

```diff
@@ -420,6 +425,20 @@ and array_set_kind =
   | Punboxedfloatarray_set of unboxed_float
   | Punboxedintarray_set of unboxed_integer
   | Punboxedvectorarray_set of unboxed_vector
+  | Pgcscannableproductarray_set of
+      modify_mode * scannable_product_element_kind list
+  | Pgcignorableproductarray_set of ignorable_product_element_kind list
+
+and ignorable_product_element_kind =
+  | Pint_ignorable
+  | Punboxedfloat_ignorable of unboxed_float
+  | Punboxedint_ignorable of unboxed_integer
+  | Pproduct_ignorable of ignorable_product_element_kind list
```

Comment:
Looks like this needs `Punboxedvector_ignorable`

Reply:
These aren't going to be supported in the first version.  The representation of the indexing in Flambda 2 needs thinking about carefully.